core/src/main/java/hudson/model/AsyncAperiodicWork.java

```diff
@@ -91,8 +138,39 @@ public void run() {
     }
 
     protected StreamTaskListener createListener() {
+        File f = getLogFile();
+        if (f.isFile()) {
+            if ((lastRotateMillis == null)
+                    || (lastRotateMillis + logRotateMillis < System.currentTimeMillis())
```

Comment:
lastRotateMillis will be null on first run so even if rotation is disabled it will be rotated once...

Is this intentional or a :big: